### PR TITLE
redux

### PR DESCRIPTION
**Pull request by DilsherB**
In this PR:
1. redux-toolkit & react-redux installed.
2. file structure created for redux implementation.
3. Two slicer created:
    a: bookSlicer with two methods inside. 1: addBook 2: removeBook
    b: categoriesSlicer with one method inside. addCategories.
4. store created with reducer to implement slicer actions.
5. Store imported and put in implementation in index.js.

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+export const categoriesSlice = createSlice({
+  name: "category",
+  initialState: [],
+  reducers: {
+    addCategory: (state, action) => {
+      state.push(action.payload);
+    },
```
Comment (Ibizugbe) [inline src/redux/categories/categoriesSlice.js]: Nice job so far😉. Kindly note that the requirement of the project states that your `categoriesSlice` Includes a reducer that checks the status and always returns "Under construction"😉.currently, you have implemented an `addCategory reducer which is not required🥲. Let's try to stick to the requirement of the project👋👍.
image

```suggestion
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },

```

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @DilsherB 👋👋, 

tryagain


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] The ducks pattern have been used for the file structure👍.
- [x] `@reduxjs/toolkit` & `react-redux` have been properly installed👍.
- [x] Gitflow have been used👍.
- [x] Linters are correctly setup and they are passing👍.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (DilsherB) [reply src/redux/categories/categoriesSlice.js]: @Ibizugbe Thank you for your review and nice explanation of the changes required.

**Review by mikethreels — APPROVED** [review]
Hi Dilsher, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions of the previous CR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**